server/src/main/java/org/elasticsearch/action/admin/indices/template/get/TransportGetComposableIndexTemplateAction.java

```diff
@@ -50,6 +54,7 @@ public TransportGetComposableIndexTemplateAction(
             GetComposableIndexTemplateAction.Response::new,
             ThreadPool.Names.SAME
         );
+        clusterSettings = DataLifecycle.isEnabled() ? clusterService.getClusterSettings() : null;
```

Comment:
I think we could have the cluster settings non-null and add the DataLifecycle.isEnabled() check only when using them (to avoid potential NPEs)